File: DeepLearningDevelopingKit/src/UnitTest/JsonHandler_test.py
#**************************************************************************************************#
#                                               Deep Learning Developing Kit                                                    #
#								        		 	          Json Handler   	                                                           #
#                                                   www.tianshicangxie.com                                                         #
#                                      Copyright © 2015-2018 Celestial Tech Inc.                                           #
#**************************************************************************************************#

# modules
import os 
import sys
import numpy as np
import json

dir_path = os.path.abspath(os.path.join(os.path.dirname(__file__),'..\\..'))

sys.path.append(str(dir_path + '\\src\\Json')) 
from JsonHandler import JsonHandler as jh
from JsonParser import VectorParser as vp

sys.path.append(str(dir_path + '\\src\\PythonAPI'))
from Vector import Vector


if __name__ == '__main__':
    test_json_path = '\\data\\test\\test2.json'
    jh.open_with_cache(test_json_path)
    vectors = vp.parse_all(jh.json_data)

    vecdict = vp.serialize(vectors)

    json_data = json.dumps(vecdict, indent=4, separators=(',', ':'))

    test_json_path = '\\data\\test\\test.json'
    jh.save_json(test_json_path, vecdict)

File: DeepLearningDevelopingKit/src/UnitTest/JsonHandler_test.cpp
/***************************************************************************************************/
/*                                               Deep Learning Developing Kit                                                   */
/*								        		 		 Json Handler Test	                                                              */
/*                                                   www.tianshicangxie.com                                                        */
/*                                      Copyright © 2015-2018 Celestial Tech Inc.                                          */
/***************************************************************************************************/

// #define JsonHandlerDebug

#ifdef JsonHandlerDebug
#include <iostream>
#include <string>
#include "..\MathLib\MathLib.h"
#include "..\Util\Json\JsonHandler.h"
#include "..\Util\Json\JsonParser.h"
#include "..\Util\Timer\Timer.h"

using namespace std;
using namespace MathLib;

int main()
{
	Util::Timer<chrono::seconds> myTimer;
	VectorParser debugParser;
	string openPath = "F:\\Software\\Top Peoject\\DeepLearningProject\\DeepLearningDevelopingKit\\DeepLearningDevelopingKit\\DeepLearningDevelopingKit\\data\\test\\test.json";
	vector<Vector<double>> tempReadVecs = debugParser.Deserialize(openPath);

	for (auto & vec : tempReadVecs)
		cout << vec << std::endl;

	Vector<double> vec1(5, MathLib::VectorType::Random);
	Vector<double> vec2(3, MathLib::VectorType::Random);
	Vector<double> vec3(10, MathLib::VectorType::Random);
	Vector<double> vec4(2, MathLib::VectorType::Random);
	Vector<double> vec5(7, MathLib::VectorType::Random);
	vector<Vector<double>> tempWriteVecs{ vec1, vec2, vec3, vec4, vec5 };

	string savePath = "F:\\Software\\Top Peoject\\DeepLearningProject\\DeepLearningDevelopingKit\\DeepLearningDevelopingKit\\DeepLearningDevelopingKit\\data\\test\\test2.json";
	debugParser.Serialize(savePath, tempWriteVecs);
	cout << "Time consume : " << myTimer.GetTime() << " ms " << std::endl;
	system("pause");
	return 0;
}
#endif // JsonHandlerDebug


